File: SimulationJava/src/Guarana.java
import javax.swing.ImageIcon;
import javax.swing.JLabel;

public class Guarana extends Roslina {

    Guarana(int xx,int yy, Swiat s){
        x=xx;
        y=yy;
        inicjatywa=0;
        sila=0;
        this.s=s;
        if(s instanceof Hexagony) obrazek = new JLabel(new ImageIcon("guaranahex.png"));
        else obrazek = new JLabel(new ImageIcon("guarana.png"));
    }

    @Override
    protected void kolizja(int xx,int yy){
        zycie = 0;
        s.setTab(x, y, s.getPole(xx, yy));
        s.setTab(xx,yy,null);
        s.getPole(x, y).setX(x);
        s.getPole(x, y).setY(y);
        s.getPole(x, y).setSila(s.getPole(x, y).getSila()+3);
        makeLog(s.getPole(x, y).wypiszNazwe(), wypiszNazwe(), 4, x, y);
    }

    @Override
    public Organizm dziecko() {
        return new Guarana(x,y,s);
    }

    @Override
    public String wypiszNazwe() {
        return "guarana";
    }

    
    @Override
    public JLabel wypisz() {
        return obrazek;
    }
    
}

File: SimulationCpp/projektPO/guarana.cpp
#include "guarana.h"
#include <iostream>
using namespace std;

Guarana::Guarana(int x, int y, Swiat* s) {
	sila = 0;
	this->x = x;
	this->y = y;
	zycie = ZYWY;
	this->s = s;
}

bool Guarana::probaRozsiania() {
	int i = rand() % SZANSANAROZSIANIEGUARANY;
	if (i == UDANA)return true;
	return false;
}

void Guarana::wypisz()const {
	SetConsoleTextAttribute(GetStdHandle(STD_OUTPUT_HANDLE), KOLORGUARANY);
	char a = SYMBOLGUARANY;
	cout << a;
}

void Guarana::kolizja(int nX, int nY) {
	zycie = MARTWY;
	makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(x), to_string(y), KOMr3);
	s->tab[x][y] = s->tab[nX][nY];
	s->tab[nX][nY] = nullptr;
	s->tab[x][y]->setX(x);
	s->tab[x][y]->setY(y);
	s->tab[x][y]->setSila(ZWIEKSZo3);
}

string Guarana::wypiszNazwe() const {
	return "guarana";
}
Organizm* Guarana::dziecko() {
	return new Guarana(x, y, s);
}
